src/middlewares/__tests__/isAuth.test.js

```diff
@@ -0,0 +1,141 @@
+const jwt = require('jsonwebtoken');
+const isAuth = require('../isAuth');
+
+const validToken = { tokenValue: 'validToken', expires: 'validTime' };
+const validTokenWithNoValidTime = {
+  tokenValue: 'invalidToken',
+  expires: 'validTime',
+};
+const invalidToken = {
+  tokenValue: 'invalidToken',
+  expires: 'validTime',
+};
+const invalidTokenWithInvalidTime = {
+  tokenValue: 'invalidToken',
+  expires: 'invalidTime',
+};
+const user = {
+  id: 1,
+  username: 'name',
+  email: 'someemail@gmail.com',
+};
+
+const res = {
+  status: jest.fn().mockReturnThis(),
+  json: jest.fn().mockReturnThis(),
+  clearCookie: jest.fn().mockReturnThis(),
+};
+const next = jest.fn().mockReturnThis();
+
+const verify = jest.spyOn(jwt, 'verify');
+
+afterAll(() => {
+  jest.clearAllMocks();
+});
+describe('isAuth function middleware', () => {
+  it('Should verify when valid token and correct secret are passed', (done) => {
+    verify.mockImplementation(() => user);
+    const req = {
+      cookies: {
+        token: validToken,
+      },
+    };
+
+    isAuth(req, res, next);
+    expect(verify).toHaveBeenCalled();
+    expect(verify).toHaveBeenCalledTimes(1);
+    expect(res.json).not.toHaveBeenCalled();
+    expect(verify).not.toThrow();
+    expect(verify).toHaveBeenCalledWith(validToken, process.env.JWT_SECRET);
+    expect(verify).toHaveReturnedWith(user);
+    done();
+  });
+
+  it('Should throw error when token time is expired', (done) => {
```

Comment:
You can rephrase this to `Should not authenticate for valid token with invalid expiry time`

Reply:
Check if other test statements also need to rephrased. Try to follow similar words throughout test statements.